### PR TITLE
added security by role for api

### PR DESCRIPTION
**Pull request by Greifelt228**

### CODE REVIEW

**Comment by Greifelt228** [comment]
https://app.asana.com/0/1202389032764573/1204039520791686/f

File: pages/api/spacehub/payment/index.ts [hunk 1]
```diff
@@ -38,8 +42,24 @@ export default async function handler(
...
+        let userPayments
+        if (user.role !== Roles.ADMIN) {
+          userPayments = payments.filter((p) => {
+            return p.payer.email === user.email
+          })
+        }
+
+        return res.status(200).json({
+          success: true,
+          data: user.role === Roles.ADMIN ? payments : userPayments,
```
Comment (mykola-element) [inline pages/api/spacehub/payment/index.ts:61]: can be create a variable: isAdmin= user.role === Roles.ADMIN
user.role === Roles.ADMIN ==>
const  isAdmin=user.role === Roles.ADMIN

user.role !== Roles.ADMIN==> !isAdmin

File: common/components/DashboardPage/blocks/payments.tsx [hunk 2]
```diff
@@ -60,6 +56,16 @@ const PaymentsBlock: FC = () => {
       width: '15%',
       render: (date) => dateToDefaultFormat(date),
     },
+    userRole === Roles.ADMIN
```
Comment (mykola-element) [inline common/components/DashboardPage/blocks/payments.tsx:59]: isAdmin too, maybe...

File: pages/api/spacehub/payment/index.ts (hunk 1 above)
Comment (VanyaRPP) [inline pages/api/spacehub/payment/index.ts:61]: can be create a variable: isAdmin= user.role === Roles.ADMIN
user.role === Roles.ADMIN ==>
const isAdmin=user.role === Roles.ADMIN

user.role !== Roles.ADMIN==> !isAdmin

File: common/components/DashboardPage/blocks/payments.tsx (hunk 2 above)
Comment (VanyaRPP) [inline common/components/DashboardPage/blocks/payments.tsx:59]: isAdmin too, maybe...

File: common/components/DashboardPage/blocks/payments.tsx [hunk 3]
```diff
@@ -44,13 +40,13 @@ const PaymentsBlock: FC = () => {
...
+  // const filteredAdminPayments = payments if use sort on FE
+  //   ?.slice()
+  //   .sort((a, b) => (a.date < b.date ? 1 : -1))
 
-  const filteredUserPayments = userResponse?.data?.payments
-    .slice()
-    .sort((a, b) => (a.date < b.date ? 1 : -1))
+  // const filteredUserPayments = userResponse?.data?.payments
+  //   .slice()
+  //   .sort((a, b) => (a.date < b.date ? 1 : -1))
```
Comment (VanyaRPP) [inline common/components/DashboardPage/blocks/payments.tsx:50]: u can remove if u don't need this

**Pull request merged**